WalletWasabi.Fluent.Desktop/Program.cs

```diff
@@ -43,63 +39,82 @@ public static int Main(string[] args)
 
 		Logger.LogDebug($"Wasabi was started with these argument(s): {(args.Any() ? string.Join(" ", args) : "none") }.");
 
+		// Crash reporting must be before the "single instance checking".
 		try
 		{
 			if (CrashReporter.TryGetExceptionFromCliArgs(args, out var exceptionToShow))
 			{
 				// Show the exception.
 				BuildCrashReporterApp(exceptionToShow).StartWithClassicDesktopLifetime(args);
-				runGui = false;
+				return 1;
 			}
 		}
 		catch (Exception ex)
 		{
-			// Anything happens here just log it and do not run the Gui.
+			// If anything happens here just log it and exit.
 			Logger.LogCritical(ex);
-			runGui = false;
+			return 1;
 		}
 
-		Exception? exceptionToReport = null;
+		(UiConfig uiConfig, Config config) = LoadOrCreateConfigs(dataDir);
 
-		if (runGui)
-		{
-			try
-			{
-				var (uiConfig, config) = LoadOrCreateConfigs(dataDir);
+		// Start single instance checker that is active over the lifetime of the application.
+		using SingleInstanceChecker singleInstanceChecker = new(config.Network);
 
-				using SingleInstanceChecker singleInstanceChecker = new(config.Network);
-				singleInstanceChecker.EnsureSingleOrThrowAsync().GetAwaiter().GetResult();
+		try
+		{
+			singleInstanceChecker.EnsureSingleOrThrowAsync().GetAwaiter().GetResult();
+		}
+		catch (Exception ex) when (ex is InvalidOperationException or OperationCanceledException)
```

Comment:
Check EnsureSingleOrThrowAsync. If we are able to signal the other application that window will pop up and the user will think he started the application - no crash report in that case. The exception thrown there is OperationCanceledException. 
In any other exception, it will throw InvalidOperationException and in that case crash report is required.

Reply:
https://github.com/zkSNACKs/WalletWasabi/issues/8252